### ISSUE TITLE
Update HTML structure of items in load window

### ISSUE THREAD

**Issue by ggeisler** [ProjectMirador/mirador#2026]
The current load window (add resource window) is, conceptually, a list of resources but is implemented as a series of `<div>` elements. Instead, we think these items should be part of an unordered list, which should enable us to provide better accessibility and visual design.

@jkeck @cbeer Was there a specific reason this was done using a `<div>` for each resource item rather than a list of `<li>`s? Asking in case I'm overlooking something or if I need to provide more details for this to be actionable.

One decision when switching to a list is how to visually separate each item in the list. The [MUI guidelines](https://material.io/develop/web/components/lists/) suggest using either  `<li role="separator" class="mdc-list-divider"></li>` or `<hr class="mdc-list-divider">` to separate items. The former is not ideal for a11y (for at least some screenreaders, each separator item is counted as an actual item in the list, so the "x items in list" count is incorrect). So as ugly as it feels, maybe we can start with using `<hr class="mdc-list-divider">` between each item and see how that looks and feels. (I'd rather use CSS to add a `border-bottom` to each item but I'm not sure how much we're trying to avoid writing custom CSS for things.)

We can fine-tune the a11y behavior and visual design as necessary after we see how the switch to an ordered list looks.
